src/App.tsx

```diff
@@ -15,48 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getSortedGoods(goods: string[], sortField: SortType): string[] {
+  const goodsCopy = [...goods];
+
+  if (sortField !== SortType.Default) {
+    goodsCopy.sort((a, b): number => {
+      switch (sortField) {
+        case SortType.Alphabet: {
+          return a.localeCompare(b);
+        }
+
+        case SortType.Length: {
+          return a.length - b.length;
+        }
+
+        default: {
+          return 0;
+        }
+      }
+    });
+  }
+
+  return goodsCopy;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.Default);
+  const isResetButtonActive = sortType || isReversed;
+
+  const handleReset = () => {
+    setIsReversed(false);
+    setSortType(SortType.Default);
+  };
+
+  const filteredGoods = getSortedGoods(goodsFromServer, sortType);
+
+  if (isReversed) {
+    filteredGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortType(SortType.Alphabet)}
+          className={cn(
+            'button', 'is-info', { 'is-light': sortType !== SortType.Alphabet },
+          )}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          onClick={() => setSortType(SortType.Length)}
+          className={cn(
+            'button', 'is-success',
+            { 'is-light': sortType !== SortType.Length },
+          )}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
```suggestion
          onClick={() => setIsReversed(isReversed => !isReversed)}
```

When we change state based on previous value, we need to use callback